src/main/webapp/js/add-deal.js

```diff
@@ -6,15 +6,34 @@ reader.onload = function(e) {
   $('#img-preview').attr('src', e.target.result);
 };
 $('#img-input').change((event) => {
+  $('#img-preview').hide();
+  $('#image-extension-message').hide();
   const input = event.currentTarget;
+  if (!extensionIsValid(input.value)) {
+    $('#image-extension-message').show();
+    input.value = '';
+  }
   if (input.files && input.files[0]) {
     reader.readAsDataURL(input.files[0]);
     $('#img-preview').show();
-  } else {
-    $('#img-preview').hide();
   }
 });
 
+/**
+ * check if filename has image extension
+ * @param {string} filename
+ * @return {boolean}
+ */
+function extensionIsValid(filename) {
+  const validExtension = ['.png', '.jpg', '.jpeg', '.gif'];
```

Comment:
I managed to upload an image with extension .bmp. Maybe should not validate image file extension here?

Reply:
should i just add bmp to the list then? Because .webp works on the frontend on the img preview, but the backend throws an error, and is quite annoying when i type everything nicely but the server tells me "bad request". I think the frontend should also at least do some basic validation so users can't submit .txt or .pdf files or something.

Reply:
I'm not sure...